net/Socket.hpp

```diff
@@ -1257,7 +1263,8 @@ class StreamSocket : public Socket,
             if (_socketHandler)
                 _socketHandler->onDisconnect();
         }
-        if (!isShutdown())
+
+        if (isOpen())
```

Comment:
We won't warn below if `onDisconnect()` decides to `closeFD()` instead of `shutdown()`.